proxy/src/ctx/transport.rs

```diff
@@ -27,6 +28,20 @@ pub struct Client {
     pub proxy: Arc<ctx::Proxy>,
     pub remote: SocketAddr,
     pub metadata: destination::Metadata,
+    pub tls_status: TlsStatus,
+}
+
+/// Identifies whether or not a connection was secured with TLS,
+/// and the reason why not.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+pub enum TlsStatus {
+    /// The TLS handshake was successful.
+    Success,
+    /// TLS was not enabled for this connection.
+    Disabled,
+    /// TLS was enabled for this connection, but we have no valid
+    /// config.
+    NoConfig,
```

Comment:
In the very near future we'll need to extend this with "We tried TLS but it didn't work so we fell back to plaintext on this connection attempt."